Source/src/core/GameObject.cpp

```diff
@@ -252,6 +252,22 @@ void Hachiko::GameObject::SetActive(bool set_active)
     {
         child->SetActive(set_active);
     }
+
+    for (Component* component : components)
```

Comment:
It feels weird to have component specific behavior and checks here. I would check the gameobject status from the light side and only use them if they are active && the gameobject is active

Reply:
I have to agree with @Bernatmago on this one as well, we need to refactor out the existing component specific logic in the game object, not add more